core/src/queue.rs

```diff
@@ -66,61 +70,68 @@ pub enum Error {
 
 impl Queue {
     /// Makes queue from configuration
-    pub fn from_configuration(cfg: &Configuration, wsv: Arc<WorldStateView>) -> Self {
+    pub fn from_configuration(cfg: &Configuration) -> Self {
```

Comment:
Actually, here I'd have to agree with @SamHSmith's design. This doesn't have to be a reference to **the** world state view, it could be a working copy. This might become useful later when the world state view is a sparse view structure that lazily computes the data.

Reply:
ok, I see